src/bin/setup/android/index.ts

```diff
@@ -15,23 +15,35 @@ import { PLATFORMS } from 'turtle/constants';
 import logger from 'turtle/logger';
 
 const which = util.promisify(_which);
+const JAVA_REQUIRED_VERSION = 8;
 const REQUIRED_TOOLS: IToolDefinition[] = [
   {
     command: 'bash',
     missingDescription: 'Please install bash',
   },
   {
     command: 'javac',
-    missingDescription: 'Please install JDK (version 8 or newer) - check https://jdk.java.net/',
+    missingDescription: `Please install JDK (version ${JAVA_REQUIRED_VERSION}) - check https://jdk.java.net/`,
     testFn: async () => {
-      const { status, stdout } = await ExponentTools.spawnAsyncThrowError(
-        'javac',
+      const { status, stdout, stderr } = await ExponentTools.spawnAsyncThrowError(
+        'java',
         ['-version'],
         { stdio: 'pipe' },
       );
       if (stdout.match(/no java runtime present/i)) {
         return false;
       }
+
+      const matchResult = stderr.match(/.*version ".*\.(.*)\..*"/);
+      if (matchResult) {
+        const [, currentJavaVersion] = matchResult;
+        if (Number(currentJavaVersion) !== JAVA_REQUIRED_VERSION) {
+          throw new Error(`You're using a wrong Java version, please install version ${JAVA_REQUIRED_VERSION}`);
+        }
+      } else {
+        logger.warn(`Couldn't find Java version number, assuming you're on Java ${JAVA_REQUIRED_VERSION}...`);
```

Comment:
I would say that if we couldn't match the string that we know should occur on a valid JDK we are less than 50% sure that this will be the Java we expect, but maybe it's ok to use this best effort here.

Reply:
I made it this way because I'm afraid that there are some exotic java versions that print the version differently but it's still Java 8. 
P.S. I don't know a lot about Java :D